build_tools/third_party/sdl2/BUILD.overlay

```diff
@@ -126,16 +131,19 @@ sdl_copts = select({
 
 sdl_linkopts = select({
     "@bazel_tools//src/conditions:windows": [
-        "-Wl,advapi32.lib",
-        "-Wl,gdi32.lib",
-        "-Wl,imm32.lib",
-        "-Wl,ole32.lib",
-        "-Wl,oleaut32.lib",
-        "-Wl,opengl32.lib",
-        "-Wl,shell32.lib",
-        "-Wl,user32.lib",
-        "-Wl,version.lib",
-        "-Wl,winmm.lib",
+        "-DEFAULTLIB:user32",
+        "-DEFAULTLIB:gdi32",
+        "-DEFAULTLIB:winmm",
+        "-DEFAULTLIB:imm32",
+        "-DEFAULTLIB:ole32",
+        "-DEFAULTLIB:oleaut32",
+        "-DEFAULTLIB:version",
+        "-DEFAULTLIB:uuid",
+        "-DEFAULTLIB:shell32",
+        "-DEFAULTLIB:advapi32",
+        "-DEFAULTLIB:hid",
+        "-DEFAULTLIB:setupapi",
+        "-DEFAULTLIB:opengl32",
```

Comment:
we shouldn't need opengl32 o_o

Reply:
There are also some defines in the config file that we could prune, like

```
/* Enable OpenGL support */
#ifndef SDL_VIDEO_OPENGL
#define SDL_VIDEO_OPENGL 1
#endif
#ifndef SDL_VIDEO_OPENGL_WGL
#define SDL_VIDEO_OPENGL_WGL 1
#endif
#ifndef SDL_VIDEO_RENDER_OGL
#define SDL_VIDEO_RENDER_OGL 1
#endif
#ifndef SDL_VIDEO_RENDER_OGL_ES2
#define SDL_VIDEO_RENDER_OGL_ES2 1
#endif
#ifndef SDL_VIDEO_OPENGL_ES2
#define SDL_VIDEO_OPENGL_ES2 1
#endif
#ifndef SDL_VIDEO_OPENGL_EGL
#define SDL_VIDEO_OPENGL_EGL 1
#endif
```

I'm a little hesitant to touch this while it is working though :P

Reply:
Removed the opengl32 lib and switched those defines to 0.